lib/panoptes_client.rb

```diff
@@ -0,0 +1,43 @@
+require 'authenticator'
+require 'panoptes/client'
+
+class PanoptesClient
+  attr_accessor :client
+
+  def initialize(token)
+    @token ||= token
+    client
+  end
+
+  def roles(user_id)
+    { }.tap do |roles|
+      response = get_roles(user_id)
+      response['project_roles'].map do |role|
+        project_id = role['links']['project'].to_i
+        roles[project_id] ||= []
+        roles[project_id] |= role['roles']
+      end
+    end
+  end
+
+  def project(slug)
+    response = client.get('projects', { slug: slug, cards: true })['projects'].first
+    { id: response['id'].to_i, slug: response['slug'] }
+  end
+
+  def env
+    Rails.env.production? ? :production : :staging
+  end
+
+  def client
+    @client ||= Panoptes::Client.new({
+      env: env,
+      public_key_path: Authenticator.key_path,
```

Comment:
The underlying client can handle this for you, https://github.com/zooniverse/panoptes-client.rb/blob/3bd483c20a94a70fac438de0fbe0661faa9b34cb/lib/panoptes/client.rb#L55

Most likely YAGNI but is there a user case where i'd like to test the tove staging app against the production API, i.e. configure these via a query param?

Reply:
I like the idea of the public key handled by the gem. If this line can just be scrapped, I'll remove it.

Reply:
This will happen with the Authenticator removal refactor.